### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by mediaformat**
Update Zap
-Federates over ActivityPub
-No longer seems to be using semver, but dates.

### CODE REVIEW

**Comment by mediaformat** [comment]
Only tested on a Debian 9 Yunohost install.
Let me know if there is anything else to test, or check.

**Comment by mediaformat** [comment]
Best to hold off on the PR, a security update was [announced](https://z.macgirvin.com/channel/mike?mid=b64.aHR0cHM6Ly96Lm1hY2dpcnZpbi5jb20vaXRlbS9mYmE2Y2RiYi0xMWQ4LTRlNmEtYWY3YS05MTcwY2YyYmI5MGQ)
I’ll test it and add to this PR

**Comment by mediaformat** [comment]
@anmol26s good to go!

**Comment by anmol26s** [comment]
We have to improve the restore script too.
Giving  777 $final_path/cache is required ?

**Comment by mediaformat** [comment]
Ok, I'll test the restore script, and try tweaking the permissions.

**Comment by mediaformat** [comment]
I tried changing the permissions on _/cache/smarty3_ but below 777 didn't work.

**Comment by mediaformat** [comment]
One thing I'm not sure about is that the restore script relies on hard-coded php-fpm version 
[Line:84](https://github.com/mediaformat/zap_ynh/blob/testing/scripts/restore#L84)
[Line:126](https://github.com/mediaformat/zap_ynh/blob/testing/scripts/restore#L126)

**Comment by anmol26s** [comment]
I have updated the sources and fixed few problem.

File: manifest.json [hunk 1]
```diff
@@ -3,9 +3,9 @@
     "id": "zap",
     "packaging_format": 1,
     "description": {
-        "en": "Zap is a full featured social network application with identity/content mirroring across multiple servers (nomadic identity) but no federate with non-nomadic servers."
+        "en": "Zap is a full featured social network application with identity/content mirroring across multiple servers (nomadic identity) federating over ActivityPub and Zot6 protocols."
     },
-    "version": "2.6~ynh1",
+    "version": "20.04.27~ynh1",
```
Comment (alexAubin) [inline manifest.json:8]: Can you confirm this is the real version number used on the upstream ?

File: scripts/backup [hunk 2]
```diff
@@ -24,68 +24,70 @@ ynh_abort_if_errors
 #=================================================
 # LOAD SETTINGS
 #=================================================
-ynh_print_info "Loading installation settings..."
+ynh_script_progression --message="Loading installation settings..." --time --weight=1
```
Comment (alexAubin) [inline scripts/backup:27]: Note that the --time here is only meant for calibrating the --weight to a different value than 1 (c.f. the progressbar thing) and should be removed after ... Putting `--time --weight=1` is pointless and leads to weird time duration being append during install by actual users...

File: scripts/upgrade [hunk 3]
```diff
@@ -80,62 +85,65 @@ fi
...
-sudo chmod -R 777 $final_path/store
-sudo mkdir $final_path/addon
-ynh_setup_source "$final_path/addon" "app_addons"
+cp -a "$tmpdir/store" "${final_path}"
+cp -a "$tmpdir/.htconfig.php" "${final_path}"
+cp -a "$tmpdir/php.log" "${final_path}"
+cp -a "$tmpdir/cache" "${final_path}"
+rm -Rf "$tmpdir"
+chmod -R 777 $final_path/store
+chmod -R 777 $final_path/cache
```
Comment (alexAubin) [inline scripts/upgrade:94]: My god ... please no chmod 777 ... I know this is from the previous code but what the hell is this needed for ...

Comment (anmol26s) [reply scripts/upgrade:94]: What would you suggest? It did not worked last time.

File: manifest.json (hunk 1 above)
Comment (mediaformat) [reply manifest.json:8]: @alexAubin yes, I just confirmed it with the developer.

File: manifest.json [hunk 4]
```diff
@@ -3,9 +3,9 @@
     "id": "zap",
     "packaging_format": 1,
     "description": {
-        "en": "Zap is a full featured social network application with identity/content mirroring across multiple servers (nomadic identity) but no federate with non-nomadic servers."
+        "en": "Zap is a full featured social network application with identity/content mirroring across multiple servers (nomadic identity) federating over ActivityPub and Zot6 protocols."
```
Comment (alexAubin) [inline manifest.json:6]:
```suggestion
        "en": "A full featured social network app with identity/content mirroring across multiple servers federating over ActivityPub and Zot6 protocols."
```

That description is super-long ... here's an attempt to reduce it ..

**Comment by anmol26s** [comment]
This maybe outdated now.

**Comment by mediaformat** [comment]
> This maybe outdated now.

Indeed, I updated to the latest ynh version that was pushed a few weeks ago.

Any plans to add a MistPark repo?

**Comment by anmol26s** [comment]
> Any plans to add a MistPark repo?

What is the difference between MistPark,Osada and Red Matrix in modern time? Maintaining so many zot6 protocol software becomes difficult.

**Comment by anmol26s** [comment]
Closing this for now. Will change the permission issue in next PR.

**Pull request closed without merging**